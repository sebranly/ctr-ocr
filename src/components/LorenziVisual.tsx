import * as React from 'react';
import { LorenziTeam } from '../types';

export interface LorenziVisualProps {
  lorenziTeams: LorenziTeam[];
  setLorenziTeams: React.Dispatch<React.SetStateAction<LorenziTeam[]>>;
}

const LorenziVisual: React.FC<LorenziVisualProps> = (props) => {
  const { lorenziTeams, setLorenziTeams } = props;

  const onChangeProperty = (index: number, isName: boolean) => (e: React.ChangeEvent<HTMLInputElement>) => {
    if (lorenziTeams.length <= index) return;

    const { value } = e.currentTarget;
    const newLorenziTeams = [...lorenziTeams];

    if (isName) newLorenziTeams[index].name = value;
    else newLorenziTeams[index].color = value;

    setLorenziTeams(newLorenziTeams);
  };

  const renderLorenziTeam = (team: LorenziTeam, index: number) => {
    const { name, color } = team;

    return (
      <div className="mb" key={index}>
        <input className="mr" onChange={onChangeProperty(index, true)} value={name} />
        <input className="ml" onChange={onChangeProperty(index, false)} type="color" value={color} />
      </div>
    );
  };

  return <div className="text-center mb2">{lorenziTeams.map(renderLorenziTeam)}</div>;
};

export { LorenziVisual };
