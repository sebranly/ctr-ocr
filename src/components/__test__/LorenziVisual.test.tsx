import * as React from 'react';
import { render } from '@testing-library/react';

import { LorenziVisual } from '../LorenziVisual';

test('LorenziVisual', () => {
  const createComponent = () => (
    <LorenziVisual
      lorenziTeams={[
        { name: 'Team 1', color: '#aaaaaa' },
        { name: 'Team 2', color: '#bbbbbb' }
      ]}
      setLorenziTeams={() => {}}
    />
  );

  const { container } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot();
});
