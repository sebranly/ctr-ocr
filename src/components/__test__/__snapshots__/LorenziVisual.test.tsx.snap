// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LorenziVisual 1`] = `
NodeList [
  <div
    class="text-center mb2"
  >
    <div
      class="mb"
    >
      <input
        class="mr"
        value="Team 1"
      />
      <input
        class="ml"
        type="color"
        value="#aaaaaa"
      />
    </div>
    <div
      class="mb"
    >
      <input
        class="mr"
        value="Team 2"
      />
      <input
        class="ml"
        type="color"
        value="#bbbbbb"
      />
    </div>
  </div>,
]
`;
